packages/@lwc/template-compiler/src/codegen/index.ts

```diff
@@ -57,15 +57,19 @@ import {
 import { SVG_NAMESPACE_URI } from '../parser/constants';
 
 function transform(codeGen: CodeGen): t.Expression {
+    const parentStack: IRNode[] = [];
+
     function transformElement(element: IRElement): t.Expression {
         const databag = elementDataBag(element);
         let res: t.Expression;
 
+        parentStack.push(element);
```

Comment:
Instead of wrapping all the `transformChildren` invocation with a stack push and a pop, I would rather recommend pushing the element to the stack directly inside `transformChildren`. This would make the code more self-contained.

```ts
function transformChildren(parent: IRNode, children: IRNode[]): t.Expression {
  // ...
  parentStack.push(parent);
  while ((current = childrenIterator.next()) && !current.done) {
    // ...
  }
  parentStack.pop();
  // ...
}
```